src/pages/libs/db.ts

```diff
@@ -1,65 +1,134 @@
+  const protocols = (
+    (rawProtocols["protocols"]?.map((x: any) => ({
+      url: x.url,
+      tvl: x.tvl || 0,
+    })) ?? []) as Protocol[]
+  ).filter((x) => x.tvl >= PROTOCOL_TVL_THRESHOLD);
+  const protocolDomains = protocols
+    .map((x) => {
+      try {
+        if (!x.url) return null;
+        return new URL(x.url).hostname.replace("www.", "");
+      } catch (error) {
+        console.log("updateDomainDbs", "error", error);
+        return null;
+      }
+    })
+    .filter((x) => x !== null)
+  console.log("updateDomainDbs", "protocolDomains", protocolDomains.length);
+  const metamaskLists = (await fetch(METAMASK_LIST_CONFIG_API).then((res) => res.json())) as {
+    fuzzylist: string[];
+    whitelist: string[];
+    blacklist: string[];
+  };
+  const metamaskFuzzyDomains = metamaskLists.fuzzylist;
+  const metamaskAllowedDomains = metamaskLists.whitelist;
+  const metamaskBlockedDomains = metamaskLists.blacklist;
+  const rawDefillamaDirectory = (await fetch(DEFILLAMA_DIRECTORY_API).then((res) => res.json())) as {
+    version: number;
+    whitelist: string[];
+    blacklist?: string[];
+    fuzzylist?: string[];
+  };
+  const defillamaDomains = rawDefillamaDirectory.whitelist;
+  const defillamaBlockedDomains = rawDefillamaDirectory.blacklist ?? [];
+  const defillamaFuzzyDomains = rawDefillamaDirectory.fuzzylist ?? [];
+  const allowedDomains = getUniqueItems(metamaskAllowedDomains, protocolDomains, defillamaDomains, ['x.com'])
+  console.log("allowedDomainsDb", allowedDomains.length);
 
-export const allowedDomainsDb = new AllowedDomainsDb();
+  const blockedDomains = getUniqueItems(metamaskBlockedDomains, defillamaBlockedDomains)
+  console.log("blockedDomainsDb", blockedDomains.length);
 
-export class FuzzyDomainsDb extends Dexie {
-  domains!: Table<Domain>;
+  const fuzzyDomains = getUniqueItems(metamaskFuzzyDomains, protocolDomains, defillamaDomains, defillamaFuzzyDomains)
+  console.log("fuzzyDomainsDb", await fuzzyDomains.length);
 
-  constructor() {
-    super("FuzzyDomainsDb");
-    this.version(1).stores({
-      domains: "domain",
-    });
+  console.timeEnd(cacheKey)
+  return {
+    allowedDomains,
+    blockedDomains,
+    fuzzyDomains,
   }
 }
 
-export const fuzzyDomainsDb = new FuzzyDomainsDb();
+function getUniqueItems(...arrays) {
+  const allItems = arrays.flat()
+  return [...new Set(allItems)]
 
-export class BlockedDomainsDb extends Dexie {
-  domains!: Table<Domain>;
+}
+
+async function updateDb() {
+  const {
+    allowedDomains = [],
+    blockedDomains = [],
+    fuzzyDomains = [],
+  } = await fetchData({
+    key: cacheKey,
+    updateFrequency: 60 * 60 * 4, // update every 4 hours
+    getData,
+  })
+  allowedDomainsDb.data = new Set(allowedDomains)
+  blockedDomainsDb.data = new Set(blockedDomains)
+  fuzzyDomainsDb.data = fuzzyDomains
+}
 
-  constructor() {
-    super("BlockedDomainsDb");
-    this.version(1).stores({
-      domains: "domain",
-    });
+updateDb()
```

Comment:
running this directly seems dangerous, because if someone ever imports a function on this file to a page such as the popup page, it will run every time the popup is opened

in other words, its very easy for new devs to mess up here and end up with performance issues, putting this behind some callback such as extension startup would be better